aether-ui/aether/ui/api/utils.py

```diff
@@ -216,167 +188,130 @@ def is_object_linked(kernel_refs, object_name, entity_type_name=''):
+        outcome['ids']['mapping'][contract.name] = contract.kernel_refs['mappings']
     else:
-        get_by_name = kernel_data_request(f'mappings/?name={pipeline.name}')['results']
+        get_by_name = kernel_data_request(f'mappings/?name={contract.name}')['results']
         if len(get_by_name):
-            outcome['exists'].append({pipeline.name: 'Pipeline (mapping) with name {} exists on kernel.'.format(
-                                    pipeline.name)})
+            outcome['exists'].append({contract.name: 'Pipeline (mapping) with name {} exists on kernel.'.format(
+                                    contract.name)})
+            outcome['ids']['mapping'][contract.name] = get_by_name[0]['id']
+
+    if pipeline.mappingset:
+        try:
+            mappingset = kernel_data_request(f'mappingsets/{pipeline.mappingset}/')
+            if json.dumps(mappingset['input'], sort_keys=True) != json.dumps(pipeline.schema, sort_keys=True):
+                outcome['exists'].append({pipeline.name: 'Input data will be changed'})
+        except Exception as e:
+            print(str(e))
     return outcome
 
 
-def publish_pipeline(pipeline, projectname, overwrite=False):
+def publish_pipeline(pipeline, projectname, contract, objects_to_overwrite={}):
     '''
-    Transform pipeline to kernel data and publish
+    Transform pipeline and contract to kernel artefacts and publish
     '''
-    outcome = {
-        'successful': [],
-        'error': [],
-        'exists': []
-    }
-    pipeline.kernel_refs = pipeline.kernel_refs if pipeline.kernel_refs else {}
-    project_data = {
-        'revision': str(uuid.uuid4()),
-        'name': projectname,
-        'salad_schema': '[]',
-        'jsonld_context': '[]',
-        'rdf_definition': '[]'
-    }
+    project_id = str(uuid.uuid4())
+    project_name = 'Aux'
+    mappingsets = []
+    mappings = []
+    schemas = []
+    outcome = {}
     try:
-        create_new_kernel_object('project', pipeline, project_data)
-        outcome['successful'].append('{} project created'.format(projectname))
-    except Exception:
-        if overwrite and 'project' in pipeline.kernel_refs:
-            update_kernel_object('project', pipeline.kernel_refs['project'], project_data)
-            outcome['successful'].append('{} project updated'.format(projectname))
-        else:
-            get_by_name = kernel_data_request(f'projects/?name={projectname}')['results'][0]
-            pipeline.kernel_refs['project'] = get_by_name['id']
-            outcome['successful'].append('Existing {} project used'.format(projectname))
-
-    for entity_type in pipeline.entity_types:
-        schema_name = entity_type['name']
-        schema_data = {
-            'revision': str(uuid.uuid4()),
-            'name': schema_name,
-            'type': entity_type['type'],
-            'definition': entity_type
-        }
+        projects = kernel_data_request(f'projects/?name=Aux')['results']
```

Comment:
If there are already `kernel_refs` values, why aren't you using them instead of assigning new ones?
I mean here and in the code below.